src/js/embed.js

```diff
@@ -59,51 +60,106 @@ function getQueryParams() {
+            if (currentRow.id === rowId) {
+                return currentRow;
             }
-            const rowData = preprocess(row);
-            const templateData = {
-                data: rowData,
-                trackingCode: {
-                    like: `${trackingCode}__like`,
-                    dislike: `${trackingCode}__dislike`,
-                    more: `${trackingCode}__more`,
-                    prev: `${trackingCode}__prev`,
-                    next: `${trackingCode}__next`,
-                },
-            };
-            render(templateFn, templateData, parentEl);
-            postRender(rowData);
-        },
-    });
+
+            return prev;
+        }, null);
+
+        if (!row) {
+            throw new Error(`row with id ${rowId} not found`);
+        }
+
+        return row;
+    }
+
+    function buildTemplateData(rowData, trackingCode) {
+        return {
+            data: rowData,
+            trackingCode: {
+                like: `${trackingCode}__like`,
+                dislike: `${trackingCode}__dislike`,
+                more: `${trackingCode}__more`,
+                prev: `${trackingCode}__prev`,
+                next: `${trackingCode}__next`,
+            },
+        };
+    }
+
+    function getRows(response) {
+        const rows = response.sheets.content;
+
+        if (!rows || !rows.length) {
+            throw new Error(`bad JSON response: ${JSON.stringify(response)}`);
+        }
+
+        return rows;
+    }
+
+    function getFormat(row) {
+        const format = formats[row.format];
+
+        if (!format) {
+            throw new Error(`format ${row.format} is not valid`);
+        }
+
+        return format;
+    }
+
+    function doRender(row, trackingCode) {
+        const format = getFormat(row);
+        const { template, postRender, preprocess } = format;
+        const templateFn = dot.template(template, null, { feedback });
+        const rowData = preprocess(row);
+        const templateData = buildTemplateData(rowData, trackingCode);
+
+        render(templateFn, templateData, parentEl);
+        postRender(rowData);
+    }
+
+    function renderTailoredAtom(err, [spreadsheetRes, tailorRes]) {
+        if (err) {
+            throw err;
+        }
+        const rows = getRows(spreadsheetRes);
+        const level = tailorRes.level || params.default;
+        const id = params[level];
+        const row = getRowById(rows, id);
+        const trackingCode = `brexit__${level}__${id}__tailored`;
+
+        doRender(row, trackingCode);
+    }
+
+    function renderUntailoredAtom(err, [spreadsheetRes]) {
+        if (err) {
+            throw err;
+        }
+        const rows = getRows(spreadsheetRes);
+        const id = params.id;
+        const row = getRowById(rows, id);
+        const trackingCode = `brexit__${row.level}__${id}__untailored`;
+
+        doRender(row, trackingCode);
+    }
+
+    const untailoredRequest = callback => callback(null, {});
```

Comment:
I feel like calling this `untailoredRequest` is a bit misleading when you glance at it, since no request is actually made. Maybe just an inline no-op function on line 161 would be clearer?